### PR TITLE
feat[en]: rule-based english paragragh join

### PR DESCRIPTION
**Pull request by voidf**
针对英文写的一些规则，实现的功能有：
- 去除每页页码和文件标识符
- 去除每个文件开头的会议时间、地点、语言、主席名以及一些重复模式
- 将行首带标号（1. 2.或者是• ）的段落特判并且将同段内容合并起来
- 将满足一些规则的行后的换行符去掉，见下表：
根据观察，有至少三个因素影响一行结尾的回车能不能被删掉
1. 次行首字母是不是小写字母
2. 本行末尾字符是不是句号
3. 本行是不是约有50个字符

### CODE REVIEW

File: alignment/en.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
+from collections import deque
+import itertools
+from datasets import load_dataset
+import re
+import datetime
+
+now_timer = datetime.datetime.now()
+
+"""
```
Comment (liyongsea) [inline alignment/en.py:9]: - docstring """
- inline comment #

File: alignment/en.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+dataset = load_dataset("ranWang/test_pdf_data", split='new')
+
+filtered_file = []
+
+EDIT_DISTANCE_THRESOLD = 3
+
+def edit_distance(s1, s2):
+    """chatgpt帮我写的n方编辑距离算法"""
```
Comment (liyongsea) [inline alignment/en.py:24]: write your docstring in google style

Comment (liyongsea) [reply alignment/en.py:24]: def edit_distance(s1, s2):
    """
    Calculate the edit distance between two strings s1 and s2. The edit distance is the minimum number of
    operations (insertions, deletions, or substitutions) required to transform one string into the other.

    The function includes optimizations to prune cases where the edit distance exceeds a predefined threshold
    (EDIT_DISTANCE_THRESOLD). It also includes optimizations to reduce the time complexity by calculating the
    character distance between the two strings.

    Args:
        s1 (str): The first string for which to calculate the edit distance.
        s2 (str): The second string for which to calculate the edit distance.

    Returns:
        int: The edit distance between s1 and s2. If the calculated edit distance exceeds EDIT_DISTANCE_THRESOLD,
        the function returns EDIT_DISTANCE_THRESOLD + 1.

    Example:
        >>> edit_distance("kitten", "sitting")
        3
        >>> edit_distance("flaw", "lawn")
        2
    """
    # Function implementation

Comment (liyongsea) [reply alignment/en.py:24]: import Levenshtein

s1 = "kitten"
s2 = "sitting"
edit_distance = Levenshtein.distance(s1, s2)
print(edit_distance)  # Output: 3

File: alignment/en.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    "从开头开始读一段连续的数字"
+    x = 0
+    for c in s:
+        if c.isdigit():
+            x = x * 10 + int(c)
+        else:
+            return x
+    return x
+
+truncate_token = 'This record contains the text of speeches delivered in English and of the '
```
Comment (liyongsea) [inline alignment/en.py:72]: - global variable should be capitalized
- try to put all global variable in one place

File: alignment/en.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        while line_marker and line_marker[-1] < lineid - 1:
+            line_marker.pop()
+
+        if line_marker and lineid - 1 == line_marker[-1]:
+            line_marker.pop()
+            outputs[-1] += ' ' + line
+        else:
+            outputs.append(line)
+
+    """根据观察，有至少三个因素影响一行结尾的回车能不能被删掉
```
Comment (liyongsea) [inline alignment/en.py:128]: inline comment please use #

File: alignment/en.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+
+        if not matched:
+            unmatched.append(line)
+            while len(unmatched) > 1:
+                filtered_page.append(unmatched.popleft())
+    while unmatched: filtered_page.append(unmatched.popleft())
+    return '\n'.join(filtered_page)
+
+
+INDEX_TOKEN = '...'
```
Comment (liyongsea) [inline alignment/en.py:216]: write a main function, and use if __name__ == "__main__"

File: alignment/en.py [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+
+
+
+if __name__ == "__main__":
+    now_timer = datetime.datetime.now()
+
+    dataset = load_dataset("ranWang/test_pdf_data", split='new')
+    filtered_file = []
+
+    for rowid, row in enumerate(dataset):
```
Comment (liyongsea) [inline alignment/en.py:252]: consider using dataseet.map

**Comment by voidf** [comment]
1. 去噪(肉眼看，写规则)
2. segmentation（中/英）
3. 让chatgpt给一个对齐的样例来评判
4. 交给bertalign

**Comment by voidf** [comment]
可能用得上的文章：
- 成段（成句）：https://centre-borelli.github.io/ruptures-docs/examples/text-segmentation/

**Review by Wzixiao — APPROVED** [review]

**Pull request merged**